framework/src/meshgenerators/ElementGenerator.C

```diff
@@ -92,3 +94,8 @@ ElementGenerator::generate()
 
   return dynamic_pointer_cast<MeshBase>(mesh);
 }
+
+void
+ElementGenerator::generateData()
+{
```

Comment:
mind a quick comment here on why you need this?

Reply:
also empty routines go in the header

Reply:
This basically tell MeshGeneratorSystem that the mesh generator is compatible with data_only mode. The presence of this method is needed to indicate that this mesh generator supports generation in data only mode